### PR TITLE
Replace ga_key with gtm_enabled config option.

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/31/
Test PASSed.

**Comment by jaycee** [comment]
QA:

* Checkout or update the `develop` branch of http://github.com/juju/juju-gui
* Bootstrap the environment of your choice
* Build the charm and deploy it
```
JUJU_GUI_BRANCH=/path/to/your/juju-gui/repository make clean package deploy
```
* Confirm the gui deploys normally
* Open /config.js on the gui and check that the `GTM_enabled` value is `false`
* Set gtm to true
```
juju set juju-gui gtm-enabled=true
```
* Confirm that `GTM_enabled` is `true` in /config.js

**Comment by bac** [comment]
NB: When this lands everything in mojo that references ga_key will need to be updated.

File: config.yaml [hunk 1]
```diff
@@ -191,3 +180,8 @@ options:
       Enables gzip compressed responses from the gui.
     type: boolean
     default: true
+  gtm-enabled:
+    description: |
+      Enables google tag manager tracking.
```
Comment (bac) [inline config.yaml]: 'google'.capitalize()

File: server/guiserver/apps.py [hunk 2]
```diff
@@ -106,6 +106,7 @@ def server():
         'jujugui.uuid': options.uuid,
         'jujugui.interactive_login': options.interactivelogin,
         'jujugui.gzip': options.gzip,
+        'jujugui.GTM_enabled': options.gtm,
```
Comment (bac) [inline server/guiserver/apps.py]: I find having all of 'gtm-enabled', 'gtm', and 'GTM_enabled' really confusing.  The horse is out of the barn, but I'd really prefer 'gtm-enabled' in config.yaml and 'gtm_enabled' everywhere else. If that is too pedantic can we kill 'GTM_enabled' in favor of just 'gtm'?

File: server/guiserver/manage.py [hunk 3]
```diff
@@ -150,6 +150,7 @@ def setup():
     define(
         'gzip', type=bool, default=False,
         help='Enable gzip compression in the gui.')
+    define('gtm', type=bool, default=False, help='Enable google tag manager.')
```
Comment (bac) [inline server/guiserver/manage.py]: s/google/Google/

File: tests/test_backends.py [hunk 4]
```diff
@@ -184,7 +182,7 @@ def test_start_insecure_ws_secure(self):
             config['read-only'], config['charmworld-url'],
             config['charmstore-url'],
             secure=True, sandbox=config['sandbox'],
-            cached_fonts=config['cached-fonts'], ga_key=config['ga-key'],
+            cached_fonts=config['cached-fonts'],
             juju_core_version=config['juju-core-version'],
```
Comment (bac) [inline tests/test_backends.py]: It looks like there is not test showing gtm gets passed through.

File: server/guiserver/apps.py (hunk 2 above)
Comment (jaycee) [inline server/guiserver/apps.py]: So...I can get you partway there. I can do gtm_enabled everywhere except the jujugui.GTM_enabled bit, and in the tornado start script where underscores are an issue. That can be your pick of 'gtm' or 'gtmenabled'.

**Comment by bac** [comment]
Hi @jaycee :+1: with some small suggestions and a request for a test.  No QA but will do later.

File: config.yaml (hunk 1 above)
Comment (jaycee) [inline config.yaml]: Done.

File: server/guiserver/manage.py (hunk 3 above)
Comment (jaycee) [inline server/guiserver/manage.py]: Done.

File: server/guiserver/apps.py (hunk 2 above)
Comment (jaycee) [inline server/guiserver/apps.py]: Done.

File: tests/test_backends.py (hunk 4 above)
Comment (jaycee) [inline tests/test_backends.py]: Fixed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/34/
Test FAILed.

**Comment by anthonydillon** [comment]
Config matches my branch :+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui-charm/35/
Test PASSed.

**Comment by bac** [comment]
QA OK

**Comment by bac** [comment]
@jaycee If you don't object, could you make the following one-line change as a drive-by?
```
diff --git a/Makefile b/Makefile
index 73f0882..0e3bf5f 100644
--- a/Makefile
+++ b/Makefile
@@ -27,7 +27,7 @@ all: setup
 # support the juju-test plugin, which calls the executable files in
 # alphabetical order.
 .PHONY: setup
-setup: releases
+setup:
 	tests/00-setup
 	# Ensure the correct version of pip has been installed.
 	# 1.4.x - 1.9.x or 6.x or 7.x
```

**Comment by jaycee** [comment]
@bac thanks for review and qa. Per discussion in IRC we'll do the drive by in another branch. 

:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-charm-merge

**Pull request merged**